sqlglot/generator.py

```diff
@@ -235,146 +229,114 @@ class Generator:
         exp.WithJournalTableProperty: exp.Properties.Location.POST_NAME,
     }
 
-    JOIN_HINTS = True
-    TABLE_HINTS = True
-    IS_BOOL = True
-
+    # Keywords that can't be used as unquoted identifier names
     RESERVED_KEYWORDS: t.Set[str] = set()
+
     WITH_SEPARATED_COMMENTS = (exp.Select, exp.From, exp.Where, exp.With)
+
     UNWRAPPED_INTERVAL_VALUES = (exp.Column, exp.Literal, exp.Neg, exp.Paren)
 
     SENTINEL_LINE_BREAK = "__SQLGLOT__LB__"
 
+    # Autofilled
+    TIME_MAPPING: t.Dict[str, str] = {}
+    TIME_TRIE: t.Dict = {}
+    QUOTE_START = "'"
+    QUOTE_END = "'"
+    IDENTIFIER_START = '"'
+    IDENTIFIER_END = '"'
+    STRING_ESCAPE = "'"
+    IDENTIFIER_ESCAPE = '"'
+    INDEX_OFFSET = 0
+    UNNEST_COLUMN_ONLY = False
+    ALIAS_POST_TABLESAMPLE = False
+    IDENTIFIERS_CAN_START_WITH_DIGIT = False
+    NORMALIZE_FUNCTIONS: bool | str = "upper"
```

Comment:
why did you change the type to bool? leave it as string, then you don't need to check is True later down

Reply:
Because we want to support switching off normalization. This was previously done through `None`, but since this property is both set in the dialect and can be passed in as an argument, `None` can mean "not set" in the context of the constructor, in which case we use the dialect's default setting. This is why I used `False` instead of `None`, hence the `bool` type. FWIW I think `False` conveys the intent more clearly too.